RecursiveExtractor/FileEntry.cs

```diff
@@ -52,11 +52,11 @@ public FileEntry(string name, Stream inputStream, FileEntry? parent = null, bool
                 FullPath = $"{ParentPath}{Path.DirectorySeparatorChar}{name}";
             }
             var printPath = FullPath;
-            if (FullPath.Contains(".."))
+            if (FullPath.Contains("..") || FullPath.Contains(":"))
             {
                 Logger.Info("ZipSlip detected in {0}. Removing unsafe path elements and extracting.", FullPath);
                 // Replace .. for ZipSlip - https://snyk.io/research/zip-slip-vulnerability
-                FullPath = FullPath.Replace("..", "");
+                FullPath = FullPath.Replace("..", "").Replace(":", "");
```

Comment:
Can you clarify when you encounter the issue? Is this an issue when extracting the VHD to FileEntry objects or only when calling the method that writes the extracted files to disc?

If the later, I'd rather sanitize the path right before writing the files to disk, rather than changing the canonical path in the FileEntry.

Reply:
I think, attach screenshots perfectly describe what I mean.
![Capture](https://user-images.githubusercontent.com/399791/191453534-12e55830-4cb6-458d-a931-77192b55ea1a.PNG)

If you want to analyze a little more you can visit the VHD Sample Download Url: https://bazaar.abuse.ch/sample/e9a1ce3417838013412f81425ef74a37608754586722e00cacb333ba88eb9aa7/
Warning: This file includes malicious files. Therefore, don't directly open or mount. Debugging on VS is not risky.

Reply:
Yes. That clarifies, thanks. I'll have a fix tomorrow that's a bit more comprehensive using https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getinvalidfilenamechars?view=net-7.0

Reply:
Thanks. I'm looking forward to your commit.

Reply:
See #89